stat/stat.go

```diff
@@ -135,6 +139,55 @@ func (s *StatterWithTimeout) doLstat(path string, infoCh chan fs.FileInfo, errCh
 	}
 
 	info, err := s.lstat(path)
+	if err == nil {
+		stat, ok := info.Sys().(*syscall.Stat_t)
+		if ok {
+			correctZeroTimes(stat, s.defTime, path, s.lstat)
+
+			correctFutureTimes(stat, s.defTime)
+		}
+	}
+
 	infoCh <- info
 	errCh <- err
 }
+
+func correctZeroTimes(stat *syscall.Stat_t, now int64, path string, lstat LstatFunc) {
+	for (stat.Atim.Sec == 0 || stat.Mtim.Sec == 0) && path != "/" {
+		path = filepath.Dir(path)
+
+		parentStat, errr := lstat(path)
```

Comment:
This doubles the number of stat calls we do, which we mustn't do. With some kind of radical reworking you could in theory cache the stat results of directories, but I don't think it's at all worth it.

Just use the oldest of any non-zero stat.*tim value, and if they're all 0, use now.

Reply:
To make sense, perhaps atime should be set to the newest of any non zero value, and m to the oldest.

Reply:
It shouldn't result in double the number of stats call; it only does another stat if an atime or mtime is set to zero, which in our DB is less than 1000 times.

We could reduce the number of stat calls by copying copy mtime -> atime (or vice versa) if it's valid; that account for all but about a dozen or so in out database.

Reply:
Alternatively, we could just leave mtimes at 0 and set atimes to now, which should essentially just have them ignored by every following step.